dotCMS/src/main/java/com/dotcms/vanity/DefaultVanityUrlHandler.java

```diff
@@ -0,0 +1,92 @@
+package com.dotcms.vanity;
+
+import com.dotcms.content.model.VanityUrl;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.db.HibernateUtil;
+import com.dotmarketing.filters.CMSFilter;
+import com.dotmarketing.filters.CmsUrlUtil;
+import com.dotmarketing.util.InodeUtils;
+import com.dotmarketing.util.UtilMethods;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * This class implements the methods defined in the {@link VanityUrlHandler}
+ * Created by oswaldogallango on 2017-06-16.
+ */
+public class DefaultVanityUrlHandler implements VanityUrlHandler{
+
+    CmsUrlUtil urlUtil = CmsUrlUtil.getInstance();
```

Comment:
this should be private final and probably will have to provide a constructor if you want to test it out